paasta_tools/kubernetes_tools.py

```diff
@@ -501,6 +502,16 @@ def __init__(self, exception: Exception, service: str, instance: str) -> None:
 
 
 class KubeClient:
+    @functools.lru_cache()  # type: ignore
```

Comment:
think it's worth leaving a comment here re: the caching? it's probably going to be somewhat surprising for future yelpers when they stumble upon this otherwise